File: androidp1/app/src/main/java/com/preappointment1/app/ui/screens/NotificationsScreen.kt
package com.preappointment1.app.ui.screens

import android.Manifest
import android.content.pm.PackageManager
import android.os.Build
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.content.ContextCompat
import com.preappointment1.app.ui.components.*
import com.preappointment1.app.ui.theme.*
import com.preappointment1.app.worker.NotificationScheduler

private data class ReminderUi(
    val id: String,
    val title: String,
    val description: String,
    val time: String
)

private val defaultReminders = listOf(
    ReminderUi(
        id = "photo",
        title = "Daily photo",
        description = "Guided photo of the tracked area",
        time = "9:00 AM"
    ),
    ReminderUi(
        id = "checkin",
        title = "Evening check-in",
        description = "Pain level and infection questions",
        time = "7:00 PM"
    ),
    ReminderUi(
        id = "missed",
        title = "Missed routine alert",
        description = "If nothing is recorded by this time",
        time = "8:30 PM"
    )
)

@Composable
fun NotificationsScreen(
    onBack: () -> Unit,
    modifier: Modifier = Modifier
) {
    val context = LocalContext.current
    val enabledStates = remember {
        mutableStateMapOf<String, Boolean>().apply {
            defaultReminders.forEach { put(it.id, true) }
        }
    }
    var testSent by remember { mutableStateOf(false) }

    val permissionLauncher = rememberLauncherForActivityResult(
        contract = ActivityResultContracts.RequestPermission(),
        onResult = { granted ->
            if (granted) {
                NotificationScheduler.scheduleRoutineReminder(
                    context = context,
                    title = "P1 — Daily routine",
                    message = "Time for your photo and check-in.",
                    delayMinutes = 0
                )
                testSent = true
            }
        }
    )

    fun sendTestNotification() {
        val needsPermission = Build.VERSION.SDK_INT >= 33 &&
            ContextCompat.checkSelfPermission(
                context, Manifest.permission.POST_NOTIFICATIONS
            ) != PackageManager.PERMISSION_GRANTED

        if (needsPermission) {
            permissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS)
        } else {
            NotificationScheduler.scheduleRoutineReminder(
                context = context,
                title = "P1 — Daily routine",
                message = "Time for your photo and check-in.",
                delayMinutes = 0
            )
            testSent = true
        }
    }

    Scaffold(
        topBar = { LpmTopBar(title = "Notifications", onBack = onBack) },
        containerColor = White,
        modifier = modifier
    ) { padding ->
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .padding(horizontal = 20.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            item {
                Spacer(modifier = Modifier.height(8.dp))
                LpmBodyText(
                    "Reminders keep your follow-up consistent. " +
                        "Each entry strengthens the report for your doctor."
                )
                Spacer(modifier = Modifier.height(12.dp))
                Text(
                    text = "DAILY REMINDERS",
                    style = MaterialTheme.typography.labelMedium,
                    color = Gray400,
                    letterSpacing = 1.5.sp,
                    fontWeight = FontWeight.SemiBold
                )
            }

            items(defaultReminders.size) { index ->
                val reminder = defaultReminders[index]
                ReminderRow(
                    reminder = reminder,
                    enabled = enabledStates[reminder.id] == true,
                    onToggle = { enabledStates[reminder.id] = it }
                )
            }

            item {
                Spacer(modifier = Modifier.height(16.dp))
                LpmSecondaryButton(
                    text = if (testSent) "Test notification sent" else "Send a test notification",
                    onClick = { sendTestNotification() },
                    enabled = !testSent
                )
                Spacer(modifier = Modifier.height(32.dp))
            }
        }
    }
}

@Composable
private fun ReminderRow(
    reminder: ReminderUi,
    enabled: Boolean,
    onToggle: (Boolean) -> Unit
) {
    LpmCard {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 20.dp, vertical = 16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Text(
                        text = reminder.title,
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.Bold,
                        color = if (enabled) Black else Gray400
                    )
                    Spacer(modifier = Modifier.width(8.dp))
                    Text(
                        text = reminder.time,
                        style = MaterialTheme.typography.labelLarge,
                        fontWeight = FontWeight.SemiBold,
                        color = if (enabled) Gray600 else Gray400
                    )
                }
                Spacer(modifier = Modifier.height(2.dp))
                Text(
                    text = reminder.description,
                    style = MaterialTheme.typography.bodySmall,
                    color = Gray400
                )
            }
            Switch(
                checked = enabled,
                onCheckedChange = onToggle,
                colors = SwitchDefaults.colors(
                    checkedThumbColor = White,
                    checkedTrackColor = Black,
                    uncheckedThumbColor = White,
                    uncheckedTrackColor = Gray200,
                    uncheckedBorderColor = Gray200
                )
            )
        }
    }
}

File: iosp1/P1/P1/UI/Screens/NotificationsScreen.swift
import SwiftUI
import UserNotifications

private struct ReminderUi: Identifiable {
    let id: String
    let title: String
    let description: String
    let time: String
}

private let defaultReminders = [
    ReminderUi(id: "photo", title: "Daily photo", description: "Guided photo of the tracked area", time: "9:00 AM"),
    ReminderUi(id: "checkin", title: "Evening check-in", description: "Pain level and infection questions", time: "7:00 PM"),
    ReminderUi(id: "missed", title: "Missed routine alert", description: "If nothing is recorded by this time", time: "8:30 PM")
]

struct NotificationsScreen: View {
    let onBack: () -> Void
    
    @State private var enabledStates: [String: Bool] = [
        "photo": true,
        "checkin": true,
        "missed": true
    ]
    @State private var testSent = false
    
    var body: some View {
        VStack(spacing: 0) {
            LpmTopBar(title: "Notifications", onBack: onBack)
            
            ScrollView {
                VStack(alignment: .leading, spacing: 16) {
                    Text("Reminders keep your follow-up consistent. Each entry strengthens the report for your doctor.")
                        .font(.body)
                        .foregroundColor(Color(UIColor.darkGray))
                        .padding(.top, 8)
                    
                    Text("DAILY REMINDERS")
                        .font(.caption)
                        .fontWeight(.semibold)
                        .kerning(1.5)
                        .foregroundColor(.gray)
                        .padding(.top, 12)
                    
                    ForEach(defaultReminders) { reminder in
                        ReminderRow(
                            reminder: reminder,
                            isEnabled: Binding(
                                get: { enabledStates[reminder.id] ?? false },
                                set: { enabledStates[reminder.id] = $0 }
                            )
                        )
                    }
                    
                    Spacer().frame(height: 16)
                    
                    LpmSecondaryButton(
                        text: testSent ? "Test notification sent" : "Send a test notification"
                    ) {
                        sendTestNotification()
                    }
                    .disabled(testSent)
                }
                .padding(.horizontal, 20)
                .padding(.bottom, 32)
            }
        }
        .background(Color.white)
    }
    
    private func sendTestNotification() {
        UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .sound, .badge]) { granted, error in
            if granted {
                let content = UNMutableNotificationContent()
                content.title = "P1 — Daily routine"
                content.body = "Time for your photo and check-in."
                content.sound = .default
                
                let request = UNNotificationRequest(
                    identifier: UUID().uuidString,
                    content: content,
                    trigger: UNTimeIntervalNotificationTrigger(timeInterval: 3, repeats: false) // 3 seconds delay
                )
                
                UNUserNotificationCenter.current().add(request)
                
                DispatchQueue.main.async {
                    testSent = true
                }
            } else {
                print("Notification permission denied")
            }
        }
    }
}

private struct ReminderRow: View {
    let reminder: ReminderUi
    @Binding var isEnabled: Bool
    
    var body: some View {
        LpmCard {
            HStack {
                VStack(alignment: .leading, spacing: 4) {
                    HStack {
                        Text(reminder.title)
                            .font(.headline)
                            .foregroundColor(isEnabled ? .black : .gray)
                        
                        Text(reminder.time)
                            .font(.subheadline)
                            .fontWeight(.semibold)
                            .foregroundColor(isEnabled ? Color(UIColor.darkGray) : .gray)
                    }
                    
                    Text(reminder.description)
                        .font(.caption)
                        .foregroundColor(.gray)
                }
                
                Spacer()
                
                Toggle("", isOn: $isEnabled)
                    .labelsHidden()
                    .tint(.black)
            }
            .padding(.horizontal, 20)
            .padding(.vertical, 16)
        }
    }
}
